apps/client/src/routes/routes.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { describe, it, expect } from 'vitest';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { createMemoryRouter, RouterProvider } from 'react-router-dom';
+import routes  from './routes';
+
+const renderRouter = (route: string) => {
+  const router = createMemoryRouter(routes, {
```

Comment:
Why not use a browser router, like the real version, to test this? Is there a testing limitation?

Reply:
using MemoryRouter in tests is preferred to BrowserRouter since it allows you to keep your routing history independent of any external source